src/main/java/Storage.java

```diff
@@ -0,0 +1,103 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Storage class abstracts the I/O method of
+ * reading and writing tasks from a file.
+ */
+public class Storage {
+
+    private static List<Task> tasks;
+
+    /**
+     * Constructs a Storage instance.
+     */
+    public Storage() {
+    }
+
+    /**
+     * Loads the tasks from the file.
+     *
+     * @return The List of all the tasks loaded from the file.
+     * @throws IOException If an I/O error occurred.
+     */
+    public List<Task> loadFile() throws IOException {
+        tasks = new ArrayList<>();
+        Path path = Paths.get(System.getProperty("user.dir"), "data", "duke.txt");
+        if (!Files.exists(path)) {
+            Files.createDirectories(path.getParent());
+            Files.createFile(path);
+        }
+        File file = new File(path.toString());
+        Scanner sc = new Scanner(file);
+        while (sc.hasNextLine()) {
+            String[] task = sc.nextLine().split("\\|");
+            switch (task[0]) {
+            case "T":
+                Task toDo = new ToDo(task[2]);
```

Comment:
you can extract some parts of the repeated code as separate method and improve the slap here.